company.el

```diff
@@ -1230,7 +1238,12 @@ can retrieve meta-data for them."
                    (list (cons prefix
                                (company--preprocess-candidates candidates))))
              (unwind-protect
-                 (company-idle-begin buf win tick pt)
+                 (if (company-call-backend 'no-cache prefix)
+                     (company-idle-begin buf win tick pt)
+                   (let ((current-prefix (company-call-backend 'prefix))
+                         (ignore-case (company-call-backend 'ignore-case)))
+                     (when (string-prefix-p prefix current-prefix ignore-case)
+                       (company-idle-begin buf win (buffer-chars-modified-tick) (point)))))
```

Comment:
I don't think remembering the line number is, strictly speaking, enough. If the user just pasted code code (without newlines), this may result in some faulty completions. The odds are low, of course, but we err on the side of correctness, so far.

We could, however, introduce a serial "current completion" number, and see if it hasn't changed. Together with the `(string-prefix-p prefix current-prefix ignore-case)` check, it might be enough.